### DISCUSSION TITLE
The "Middleware" must first process the request.

### DISCUSSION THREAD

**Discussion by develgold** [laravel/framework#41711] [Q&A]
- Laravel Version: 9.5.1
- PHP Version: 8.1
- Database Driver & Version: Postgresql 13

### Steps To Reproduce:
api.php
```php
Route::middleware('testing')->get('/user/{user}', function (\App\Models\User $user) {
    logger()->info('User: ' . $user->name);
    return $user;
});
```
TestingMiddleware
```php
public function handle(Request $request, Closure $next)
{
    logger()->info('Testing middleware');
    return $next($request);
}
```
User
```php
protected static function boot()
{
    self::retrieved(function ($model) {
        logger()->info('Retrieved model: ' . get_class($model));
    });
    parent::boot();
}
```
Log
```
local.INFO: Retrieved model: App\Models\User  
local.INFO: Testing middleware  
local.INFO: User: John
```

**Comment by gguney** [comment]
Maybe user boot is checking the guest or some kind of auth? Or maybe since you are giving User model as $user to your route that is first booting it then setting user to John. Can you try this with another model like /posts/{post} etc.

**Reply by develgold** [reply]
> Maybe user boot is checking the guest or some kind of auth? Or maybe since you are giving User model as $user to your route that is first booting it then setting user to John. Can you try this with another model like /posts/{post} etc.

Same problem

api.php
```php
Route::middleware('testing')->get('/posts/{post}', function (\App\Models\Post $post) {
    logger()->info('Post title: ' . $post->title);
    return $post;
});
```
Post
```php
class Post extends Model
{
    protected static function boot()
    {
        self::retrieved(function ($model) {
            logger()->info('Retrieved model: ' . get_class($model));
        });
        parent::boot();
    }
}
```
Log
```
local.INFO: Retrieved model: App\Models\Post  
local.INFO: Testing middleware  
local.INFO: Post title: Testing Post  
```

**Reply by gguney** [reply]
No, I mean just change the route. Anyway I misunderstood.

 It is booting before your model of course because how else it should be? It will boot it then bind your data.

[here](https://laravel.com/docs/9.x/routing#route-model-binding) you can read more about it. If you do not want to bind it before your route just use simple URL with id.
